File: scripts/bump_version.py
#!/usr/bin/env python3
"""Bump CCR app version before git commit."""
import re
import sys
from pathlib import Path

APP_DIR = Path('/Users/willy/Projects/ccr-app')

# Find current version
app_js = APP_DIR / 'js' / 'app.js'
content = app_js.read_text()
m = re.search(r"APP_VERSION = '(\d+)\.(\d+)\.(\d+)'", content)
if not m:
    print("ERROR: Could not find APP_VERSION"); sys.exit(1)

major, minor, patch = int(m.group(1)), int(m.group(2)), int(m.group(3))
new = f"{major}.{minor}.{patch+1}"
print(f"Bumping {major}.{minor}.{patch} → {new}")

# Update app.js
content = re.sub(r"APP_VERSION = '\d+\.\d+\.\d+'", f"APP_VERSION = '{new}'", content)
app_js.write_text(content)

# Update sw.js
sw = APP_DIR / 'sw.js'
sw_content = sw.read_text()
sw_content = re.sub(r"APP_VERSION = '\d+\.\d+\.\d+'", f"APP_VERSION = '{new}'", sw_content)
sw_content = re.sub(r"ccr-app-v\d+", f"ccr-app-v{new}", sw_content)
sw.write_text(sw_content)

# Update FAB badge in index.html
ih = APP_DIR / 'index.html'
ih_content = ih.read_text()
ih_content = re.sub(r">(v\d+\.\d+\.\d+)<", f">{new}<", ih_content)
ih.write_text(ih_content)

print(f"✅ Done — all files updated to {new}")

File: scripts/bump_version.js
#!/usr/bin/env node
/** Bump CCR app version — run before git commit */
const fs = require('fs');
const path = require('path');

const dir = '/Users/willy/Projects/ccr-app';
const files = {
  app: path.join(dir, 'js/app.js'),
  sw:  path.join(dir, 'sw.js'),
  ih:  path.join(dir, 'index.html')
};

function read(f) { return fs.readFileSync(f, 'utf8'); }
function write(f, c) { fs.writeFileSync(f, c); }

let content = read(files.app);
const m = content.match(/APP_VERSION = '(\d+)\.(\d+)\.(\d+)'/);
if (!m) { console.error('Could not find APP_VERSION'); process.exit(1); }
const [, major, minor, patch] = m.map(Number);
const next = `${major}.${minor}.${patch+1}`;
console.log(`Bumping ${major}.${minor}.${patch} → ${next}`);

['app', 'sw'].forEach(k => {
  let c = read(files[k]);
  c = c.replace(/APP_VERSION = '\d+\.\d+\.\d+'/g, `APP_VERSION = '${next}'`);
  if (k === 'sw') c = c.replace(/ccr-app-v\d+/g, `ccr-app-v${next}`);
  write(files[k], c);
});

let ih = read(files.ih);
ih = ih.replace(/>v\d+\.\d+\.\d+</g, `>v${next}<`);
write(files.ih, ih);

console.log(`✅ Bumped to ${next}`);
